src/firefly/js/tables/ui/TablesContainer.jsx

```diff
@@ -87,44 +82,42 @@ ExpandedView.defaultProps = TabsView.defaultProps;
 
 
 function TabsView(props) {
-    const {tables, tbl_ui_gid, expandedMode} = props;
+    const {tables, expandedMode, active} = props;
+
+    var activeIdx = Object.keys(tables).findIndex( (tbl_ui_id) => get(tables,[tbl_ui_id,'tbl_id']) === active);
+    activeIdx = activeIdx === -1 ? 0 : activeIdx;
     const onTabSelect = (idx) => {
         const tbl_id = get(tables, [Object.keys(tables)[idx], 'tbl_id']);
         tbl_id && dispatchActiveTableChanged(tbl_id)
     };
 
     return (
-        <Tabs defaultSelected={0} onTabSelect={onTabSelect}>
-            {tablesAsTab(tables, tbl_ui_gid, expandedMode)}
+        <Tabs defaultSelected={activeIdx} onTabSelect={onTabSelect}>
```

Comment:
Looks like it fixed the issue of the selected table tab was not being preserved on expand/collapse.